scripts/auth.js

```diff
@@ -27,6 +38,102 @@ window.onload = () => {
             document.querySelector(".cart-link").childNodes[1].nodeValue = `Cart ($${userCart.cartPrice.toFixed(2)})`
         })
     } else {
-        console.log("Auth failed, no user in local storage")
+        console.log("Auth failed, no user in local storage");
     }
+
+    let apiDataRef = firebase.database().ref("api");
+    let data = [];
+    let bannerPic = document.querySelector(".banner");
+    let tagsContainer = document.querySelector(".tags-container");
+    let recipeContainer = document.querySelector(".recipe-content");
+    let categoriesContainer = document.querySelector(".cards-container")
+    //Selects data from firebase and adds the html elements accordingly
+    apiDataRef.on('value', async (snapshot) => {
+        data = await snapshot.val();
+
+        if (
+            data
+            && data.results
+            && data.results[0]
+        ) {
+
+            if (data.results[0].imageUrl) {
+                bannerPic.style.backgroundImage = `url(${data.results[0].imageUrl})`;
+            } else {
+                console.log("No data is present in db for imageUrl.");
+            }
+
+            if (tagsContainer) {
+
+                if (
+                    data.results[0].tags
+                    && data.results[0].tags.length > 0
+                ) {
+                    tagsContainer.innerHTML =
+                        `<span>
+                    <img src="assets/svg/tag.svg" class="tag-svg" alt="Tag icon"/>
+                    </span>`;
+                    data.results[0].tags.forEach(tag => {
+                        if (tag) {
+                            tagsContainer.innerHTML += `<span class="tags">${tag}</span>`;
+                        }
+                    })
+                }
+
+            } else {
+                console.log("The recipeContent container had an error.");
+            }
+
+            if (data.results[0].name
+                && tagsContainer) {
+
+                tagsContainer.insertAdjacentHTML("afterend", `<h1 class="title">${data.results[0].name}</h1>`);
+
+            }
+            if (data.results[0].description
+                && recipeContainer) {
+
+                recipeContainer.insertAdjacentHTML("beforeend",
+                    `<p class="description">${data.results[0].description}</p>`
+                );
+            }
+        }
+        let categories = [];
+        data.results.forEach((result) => {
+            if (result.tags && result.tags.length > 0) {
+                result.tags.forEach(tag => {
+                    categories.push(tag);
```

Comment:
why you don't just ES6 like: categories=[...categories, ...result.tags] ? search them